addon/mixins/store.js

```diff
@@ -276,21 +264,14 @@ function createRecordDataFor(modelName, id, clientId, storeWrapper) {
     );
   }
 
+  // TODO: what is the purpose of this check?
   if (HAS_EMBER_DATA_PACKAGE) {
```

Comment:
probably because there was nothing to super to if you don't have the meta package, although that is something we're going to fix. TL;DR a mistake (maybe rectified? possibly forgotten during my leave?) was made wherein the meta package is what configured the store to use @ember-data/model.